scripts/pysa_fuzzer/code_generator.py

```diff
@@ -0,0 +1,324 @@
+from typing import List
+import random
+import textwrap
+import itertools
+import string
+
+class CodeGenerator:
+    def __init__(self) -> None:
+        self.variables = self.generate_variable_names()
+        self.current_var = 0
+    
+    def generate_variable_names(self) -> List[str]:
+        single_letter_names = list(string.ascii_lowercase)
+        two_letter_names = [''.join(pair) for pair in itertools.product(string.ascii_lowercase, repeat=2)]
+        names = single_letter_names + two_letter_names
+        # Remove reserved keywords
+        reserved_keywords = {'as', 'in', 'if', 'is', 'or'}
+        names = [name for name in names if name not in reserved_keywords]
+        return names
+
+    def generate_new_variable(self) -> str:
+        index = self.current_var
+        self.current_var += 1
+        return self.variables[index]
+
+    def get_last_variable(self) -> str:
+        return self.variables[self.current_var - 1]
+
+    def reset(self) -> None:
+        self.current_var = 0
+
+    def generate_import_statements(self) -> str:
+        return "import random\nimport math\n"
+    
+    def generate_source(self) -> str:
+        curr_var = self.generate_new_variable()
+        return f"{curr_var} = input()"
+
+    def generate_sink(self) -> str: 
+        prev_var = self.get_last_variable()
+        return f"print({prev_var})"
+
+    def generate_addition(self, depth: int = 1) -> str:
+        prev_var = self.get_last_variable()
+        curr_var = self.generate_new_variable()
+        addition_code = f"{curr_var} = {prev_var} + '{random.randint(1, 9)}'"
+        
+        if depth > 1:
+            nested_addition = self.generate_addition(depth - 1)
+            return f"{addition_code}\n{nested_addition}"
+        else:
+            return addition_code
+
+    def generate_for_loop(self, depth: int = 1) -> str:
+        prev_var = self.get_last_variable()
+        curr_var = self.generate_new_variable()
+        loop_body = f"{curr_var} += {prev_var}"
+
+        if depth > 1:
+            nested_loop = self.generate_for_loop(depth - 1)
+            loop_body = f"{nested_loop}\n{textwrap.indent(loop_body, '    ')}"
+
+        return f"{curr_var} = ''\nfor _ in range({random.randint(2, 5)}):\n{textwrap.indent(loop_body, '    ')}"

```

Comment:
here and other places: could we do something like this (see below) so it's easier to read?
```suggestion
        return (
          f"{curr_var} = ''\n"
          + f"for _ in range({random.randint(2, 5)}):\n"
          + f"{textwrap.indent(loop_body, '    ')}")

```